hydra-cluster/src/HydraNode.hs

```diff
@@ -172,42 +173,42 @@ waitForAll tracer delay nodes expected = do
 -- | Helper to make it easy to obtain a commit tx using some wallet utxo.
 -- Create a commit tx using the hydra-node for later submission.
 requestCommitTx :: HydraClient -> UTxO -> IO Tx
-requestCommitTx HydraClient{hydraNodeId} utxos =
+requestCommitTx HydraClient{peer = Host{hostname, port}} utxos =
   runReq defaultHttpConfig request <&> commitTx . responseBody
  where
   request =
     Req.req
       POST
-      (Req.http "127.0.0.1" /: "commit")
+      (Req.http hostname /: "commit")
       (ReqBodyJson $ SimpleCommitRequest @Tx utxos)
       (Proxy :: Proxy (JsonResponse (DraftCommitTxResponse Tx)))
-      (Req.port $ 4_000 + hydraNodeId)
+      (Req.port (fromInteger . toInteger $ port))
 
 -- | Submit a decommit transaction to the hydra-node.
 postDecommit :: HydraClient -> Tx -> IO ()
-postDecommit HydraClient{hydraNodeId} decommitTx = do
+postDecommit HydraClient{peer = Host{hostname, port}} decommitTx = do
   void $
-    parseUrlThrow ("POST http://127.0.0.1:" <> show (4000 + hydraNodeId) <> "/decommit")
+    parseUrlThrow ("POST http://" <> T.unpack hostname <> ":" <> show port <> "/decommit")
       <&> setRequestBodyJSON decommitTx
         >>= httpLbs
 
 -- | Get the latest snapshot UTxO from the hydra-node. NOTE: While we usually
 -- avoid parsing responses using the same data types as the system under test,
 -- this parses the response as a 'UTxO' type as we often need to pick it apart.
 getSnapshotUTxO :: HydraClient -> IO UTxO
-getSnapshotUTxO HydraClient{hydraNodeId} =
+getSnapshotUTxO HydraClient{peer = Host{hostname, port}} =
```

Comment:
Must: must not call this `peer`, but `apiHost` or so (to avoid confusion)

It's _not_ a "peer" what we configure the hydra-node with (`--peer`), but it points to the API of a hydra node.